### ISSUE TITLE
Improved tests

### ISSUE THREAD

**Issue by Limpan** [teknik-eksjo/chronos#74]
The application lacks quite a few tests. It would be good for avoiding regressions.
- Unit tests for the model.
- Functional tests of the different blueprints.
- Possibly some selenium tests?

**Comment by Greenheart** [comment]
I've added some tests for `models.py` but there are still methods in the User-model that's untested.

@Limpan Should I add a PR with the tests I've got so far to let someone else take over? Or should they work towards [my branch on my fork](https://github.com/Greenheart/chronos/tree/feature/datamodel-tests)?

**Another question:** how do I view the coverage report? Some html-file?

**Comment by Limpan** [comment]
Make a PR, the more the merrier! We'll also be able to at least catch some regressions.

Coverage report is in `tmp/html/index.html` or something similar.

**Comment by Greenheart** [comment]
Will we be able to squash the commits from my branch into **only one** when merging? I'm not 100% of how the GitHub's `squash-n-merge`-button works :stuck_out_tongue:

**Comment by Limpan** [comment]
Yes, but it's better if you rebate and squash first. That way you can write a good commit message.

**Comment by Greenheart** [comment]
@Limpan I think it's possible to create the new commit message while doing the squash-n-merge-thingy here on GH. At least according to [this guide](https://github.com/blog/2141-squash-your-commits)

**Comment by Limpan** [comment]
Tests that are missing:

### Unit tests
- [ ] Test permissions of users with different roles, including the anonymous user.
- [ ] Test that inactive users can't do anything.
- [ ] Test user ping.
- [ ] Test user password reset

### With Flask test client
- [ ] Test that the home page loads and the main pages of all blueprints (admin and possibly auth).
- [ ] Test other relevant pages.
- [ ] Test login

### Selenium based tests
- [ ] Test some reasonable scenarios.
